python/shotgun_menus/entity_field_menu.py

```diff
@@ -221,7 +234,12 @@ def entity_filter(et):
                     bubble_fields[field] = {
                         "name": display_name,
                         "valid_types": entity_types,
-                        "valid_type_names": [shotgun_globals.get_type_display_name(et) for et in entity_types],
+                        "valid_type_names": [
+                            shotgun_globals.get_type_display_name(
+                                et,
+                                project_id=self._project_id,
+                            ) for et in entity_types
+                        ],
```

Comment:
It does seem brutal, and that's a really big chunk of logic to wrap in such an all-encompassing try/except. Unfortunately, I'm not familiar enough with that chunk of logic to feel comfortable changing it (It's Rob's/Josh's stuff).